scripts/download_technique_animations.py

```diff
@@ -24,30 +29,34 @@ def download(url, filename):
     return filename
 
 
-def process_filename(filename):
-    filename = filename[:-4]
-    filename = filename.strip("0123456789_")
-    filename = filename.lower()
-    return filename
+def process_filename(filename: str) -> str:
+    """Extract base filename from an animation file path."""
+    cleaned_filename = os.path.splitext(filename)[0]
+    cleaned_filename = cleaned_filename.strip("0123456789_")
+    cleaned_filename = cleaned_filename.lower()
+    return cleaned_filename
 
 
-def process_gif(filename):
-    image = Image.open(filename)
-    if not image.is_animated:
-        print("...not animated, skipped")
-        return
-    if not image.width == image.height == 64:
-        print("...is not 64x64, skipped")
-        return
-    base_name = process_filename(filename)
-    for frame in range(0, image.n_frames):
-        frame_filename = f"{base_name}{frame:02}.png"
-        print(f"saving {filename}:{frame_filename} {frame}/{image.n_frames - 1}")
-        image.seek(frame)
-        image.save(frame_filename, optimize=True)
+def process_gif(filename: str) -> None:
+    """Extract individual animation frames as PNG from a GIF."""
+    with Image.open(filename) as image:
```

Comment:
I'm not sure whether there is a chance to have too many image files opened, but `Image.open` is a context manager so we might use it here :)